src/burnchains/events.rs

```diff
@@ -63,6 +63,18 @@ pub struct NewBlock {
     pub events: Vec<NewBlockTxEvent>,
 }
 
+impl NewBlock {
+    /// Shortened debug string, for logging.
+    pub fn short_string(&self) -> String {
+        format!(
+            "NewBlock(hash={:?}, parent_hash={:?}, block_height={}, num_events={}",
+            &self.index_block_hash,
+            &self.parent_index_block_hash,
+            self.block_height,
+            self.events.len()
+        )
+    }
+}
```

Comment:
You should just implement Debug for this:

```suggestion
impl std::fmt::Debug for NewBlock {
    /// Shortened debug string, for logging.
    fn fmt(&self, f: &mut Formatter) -> Result<(), std::fmt::Error>
        write!(
            "NewBlock(hash={:?}, parent_hash={:?}, block_height={}, num_events={}",
            &self.index_block_hash,
            &self.parent_index_block_hash,
            self.block_height,
            self.events.len()
        )
    }
}
```